central/declarativeconfig/updater/role_updater.go

```diff
@@ -55,6 +59,24 @@ func (u *roleUpdater) DeleteResources(ctx context.Context, resourceIDsToSkip ...
 	var roleDeletionErr *multierror.Error
 	var roleNames []string
 	for _, role := range roles {
+		referencingGroups, err := u.groupDatastore.GetFiltered(ctx, func(group *storage.Group) bool {
+			return group.GetRoleName() == role.GetName()
+		})
+		if err != nil || len(referencingGroups) > 0 {
+			if err == nil {
+				err = errox.ReferencedByAnotherObject.Newf("role is still referenced by groups")
+			}
+			roleDeletionErr = multierror.Append(roleDeletionErr, err)
+			roleNames = append(roleNames, role.GetName())
+
+			u.reporter.UpdateIntegrationHealthAsync(utils.IntegrationHealthForProtoMessage(role, "", err,
+				u.idExtractor, u.nameExtractor))
+			role.Traits.Origin = storage.Traits_DECLARATIVE_ORPHANED
+			if err = u.roleDS.UpdateRole(ctx, role); err != nil {
+				roleDeletionErr = multierror.Append(roleDeletionErr, err)
+			}
```

Comment:
Same as above whether we should set this as orphaned when `errors.Is(err, errox.ReferencedByAnotherObject) == false`

Reply:
I realised `RemoveRole` actually checks for group references, so simplified :D